controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook_test.go

```diff
@@ -293,6 +291,23 @@ func TestKubeadmControlPlaneValidateUpdate(t *testing.T) {
 	scaleToEvenExternalEtcdCluster := beforeExternalEtcdCluster.DeepCopy()
 	scaleToEvenExternalEtcdCluster.Spec.Replicas = pointer.Int32Ptr(2)
 
+	beforeInvalidEtcdCluster := before.DeepCopy()
+	beforeInvalidEtcdCluster.Spec.KubeadmConfigSpec.ClusterConfiguration = &kubeadmv1beta1.ClusterConfiguration{
```

Comment:
This is testing a slightly different use case than what the issue describes. This test cases is for an object that is invalid to start with won't accept any updates.

Could you modify this test so that the before configuration has local and then the after is trying to add an external configuration?